File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        dic = collections.defaultdict(int)
        for index in range(len(nums)):
            dic[nums[index]-target] = index

        for index in range(len(nums)):
            # print(index)
            # if -1 * nums[index] not in dic:
            #     continue

            val = dic[-1 * nums[index]]
            if val != 0 and index != val:
                return [index, val]
        

File: 0001-two-sum/0001-two-sum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> numToIdx = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            numToIdx.put(nums[i], i); // if present, overwrite
        }

        for(int i = 0; i < nums.length; i++){
            Integer idx = numToIdx.get(target - nums[i]);
        
            if(idx != null && idx != i){
                return new int[]{i, idx};
            }
        }

        return new int[]{0,0}; // only to prevent compile error
    }
}